src/redash_stmo/datasource_health/health.py

```diff
@@ -0,0 +1,92 @@
+import json
+import time
+import redash
+import settings
+
+from celery.utils.log import get_task_logger
+from redash import models, redis_connection, statsd_client
+from redash.worker import celery
+from redash.utils import parse_human_time
+from redash.monitor import get_status
+from redash.query_runner import BaseQueryRunner
+
+original_get_status = get_status
+logger = get_task_logger(__name__)
+
+
+def update_health_status(data_source_id, data_source_name, query_text, data):
+    key = "data_sources:health"
+
+    cache = json.loads(redis_connection.get(key) or '{}')
+    if data_source_id not in cache:
+        cache[data_source_id] = {
+            "metadata": { "name": data_source_name },
+            "queries": {}
+        }
+    cache[data_source_id]["queries"][query_text] = data
+
+    cache[data_source_id]["status"] = "SUCCESS"
+    for query_status in cache[data_source_id]["queries"].values():
+        if query_status["status"] == "FAIL":
+            cache[data_source_id]["status"] = "FAIL"
+            break
+
+    redis_connection.set(key, json.dumps(cache))
+
+
+@celery.task(name="redash_stmo.datasource_health.health.health_status", time_limit=90, soft_time_limit=60)
+def health_status():
```

Comment:
Let's rename this to `update_health_status` and rename the `update_health_status` function above to `store_health_status`.